src/dbnode/client/options.go

```diff
@@ -877,3 +883,22 @@ func (o *options) SetFetchSeriesBlocksBatchConcurrency(value int) AdminOptions {
 func (o *options) FetchSeriesBlocksBatchConcurrency() int {
 	return o.fetchSeriesBlocksBatchConcurrency
 }
+
+func (o *options) SetAsyncTopologyInitializers(value []topology.Initializer) Options {
+	opts := *o
+	opts.asyncTopologyInitializers = value
+	return &opts
+}
+
+func (o *options) AsyncTopologyInitializers() []topology.Initializer {
+	return o.asyncTopologyInitializers
+}
+
+func (o *options) OptionsForAsyncClusters() []Options {
```

Comment:
super nit: If you were to make this an extra method it would keep the `Options` as mainly just setter/getter and validate rather than "computed properties" getters as well. 

i.e. 
```go
func NewOptionsForAsyncClusters(opts Options) []Options {
	result := make([]Options, 0, len(opts. AsyncTopologyInitializers()))
	for _, topoInit := range opts. AsyncTopologyInitializers() {
		options := opts.SetTopologyInitializer(topoInit)
		result = append(result, options)
	}
	return result
}
```